src/qt/calctrl.cpp

```diff
@@ -24,6 +24,11 @@
 #include <QtGui/QTextCharFormat>
 #include <QtWidgets/QCalendarWidget>
 
+namespace
+{
+    const int MAX_YEAR_QT = 7999;
```

Comment:
I'd add a comment saying that the native control doesn't support years beyond this one (at least if this is true, i.e. if I understand correctly what's going on here).